pkg/vdr/web/did.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"strings"
+)
+
+const (
+	defaultPath  = "/.well-known/doc.json"
+	documentPath = "/doc.json"
+)
+
+// parseDIDWeb consumes a did:web identifier and returns the URL location of the did Doc according to the
+// spec(https://w3c-ccg.github.io/did-method-web/#crud-operation-definitions) but always of the form https://.../doc.json
+func parseDIDWeb(id string) (string, error) {
+	var urlString string
+
+	// parses did to make sure it conforms to generic did standard
+	_, err := did.Parse(id)
+	if err != nil {
+		return urlString, fmt.Errorf("invalid did, does not conform to generic did standard --> %w", err)
+	}
+
+	parts := strings.Split(id, ":")
+	if parts[0] != "did" || parts[1] != "web" {
```

Comment:
You can use the result of `did.Parse()`, which is a [`did.DID`](https://github.com/hyperledger/aries-framework-go/blob/e86b70555480fba40df163f17f38ccc6d80a0525/pkg/doc/did/doc.go#L65-L70):

```go
// DID is parsed according to the generic syntax: https://w3c.github.io/did-core/#generic-did-syntax
type DID struct {
	Scheme           string // Scheme is always "did"
	Method           string // Method is the specific DID methods
	MethodSpecificID string // MethodSpecificID is the unique ID computed or assigned by the DID method
}
```